Makefile

```diff
@@ -36,7 +36,8 @@ TEST_TYPE=$(word 3, $(subst _, ,$(EXPERIMENT)))
 THRESH_ARGS=--threshold_overrides_file=$(FV3_PATH)/tests/translate/overrides/$(TEST_TYPE).yaml
 TEST_ARGS_USE=$(TEST_ARGS) $(THRESH_ARGS)
 PYTEST_SEQUENTIAL=pytest --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS_USE) $(FV3_PATH)/tests
-PYTEST_PARALLEL=$(MPIRUN_CALL) pytest --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS_USE) -m parallel $(FV3_PATH)/tests
+# we can't rule out a deadlock if one test fails, so we must set maxfail=1 for parallel tests
+PYTEST_PARALLEL=$(MPIRUN_CALL) python -m mpi4py -m pytest --maxfail=1 --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS_USE) -m parallel $(FV3_PATH)/tests
```

Comment:
We need something like this for daint tests, though there is a potential use case if you run locally and know the tests you're running won't deadlock and want to see if more than the first test fails. Right now you'd edit the Makefile and delete the maxfail temporarily to do that.

The alternative is to somehow incorporate it into one of the "passable" arguments like TEST_ARGS, but this is complicated by the fact that we want this by default for only parallel tests. So we'd have to make `TEST_ARGS_PARALLEL?= --maxfail=1 $(TEST_ARGS)` and then pass TEST_ARGS_PARALLEL from the command line to override the maxfail. Learning these two arguments for this very particular use case seems complex enough to me that it's OK to say you have to edit the Makefile for this use case. I'm willing to do it the other way if you think this use case comes up more often, though.

Reply:
go ahead and add this, I have heard of stuck tests hitting a few people since the first review of this, we can always change again if the workflow doesn't work for us. I personally seem to run into stuck parallel tests very infrequently, but if it is hitting other people we should do it. Luckily there are only a handful of parallel tests.